applications/mne_analyze/libs/anShared/Model/eventmodel.cpp

```diff
@@ -711,3 +703,88 @@ std::vector<uint> EventModel::getEventSelection() const
 {
     return m_listEventSelection;
 }
+
+//=============================================================================================================
+
+void EventModel::updateSelectedGroups(const QList<QModelIndex> &indexList)
+{
+    clearGroupSelection();
+
+    for(auto row : indexList){
+        addToSelectedGroups(row.data(Qt::UserRole).toInt());
+    }
+
+    eventsUpdated();
+}
+
+void EventModel::getEventsFromNewData()
+{
+    auto info = m_pFiffModel->getFiffInfo();
+    auto raw = m_pFiffModel->getFiffIO()->m_qlistRaw.first().data();
+
+    auto previousLastSample = m_pFiffModel->getPreviousLastSample();
+
+    int iFirstSample = m_pFiffModel->absoluteFirstSample();
+
+    std::list<int> indexList;
+
+    for(int i = 0; i < info->chs.size(); i++) {
+        if(info->chs[i].kind == FIFFV_STIM_CH) {
+            indexList.push_back(i);
+        }
+    }
+
+    Eigen::MatrixXd mSampleData, mSampleTimes;
+
+    if(previousLastSample > 0){
+        raw->read_raw_segment(mSampleData,
+                              mSampleTimes,
+                              previousLastSample);
+    }
+    else {
+        raw->read_raw_segment(mSampleData,
+                              mSampleTimes);
+    }
+
+    for (int iChannelIndex : indexList){
+        QList<QPair<int,double>> detectedTriggerSamples = RTPROCESSINGLIB::detectTriggerFlanksMax(mSampleData,
+                                                                                                  iChannelIndex,
+                                                                                                  0,
+                                                                                                  m_dThreshold,
+                                                                                                  0);
+
+        QMap<double,QList<int>> mEventsinTypes;
+
+        for(QPair<int,double> pair : detectedTriggerSamples){
```

Comment:
reference would avoid creating a variable calling QPair constructor each iteration. 
const auto& would make it const so easier to optimize for the compiler?